### DISCUSSION TITLE
A question about the LOSS function in CRLI models

### DISCUSSION THREAD

**Discussion by PBemmm** [WenjieDu/PyPOTS#292] [General]
In the tensorflow source code of paper, there are five LOSS functions (loss_d, loss_G, loss_pre, loss_re, loss_km).

 #loss_D
        loss_D = tf.nn.sigmoid_cross_entropy_with_logits(logits = disc_reshaped,labels = mask_inputs)
        loss_D = tf.multiply(loss_D,length_mark)
        loss_D = tf.reduce_sum(tf.reduce_mean(loss_D,0))
        #loss_G
        loss_G = tf.nn.sigmoid_cross_entropy_with_logits(logits = disc_reshaped,labels = 1-mask_inputs)
        loss_G = tf.multiply(loss_G,length_mark)
        loss_G = tf.multiply(loss_G,1-mask_inputs)
        loss_G = tf.reduce_sum(tf.reduce_mean(loss_G,0))
        #loss_pre
        pre_tmp = tf.multiply(impute_vector,masks)
        targ_pre = tf.multiply(masks,corpt_inputs)
        loss_pre = mse_error(pre_tmp,targ_pre) 
        #loss_re
        output_trim = decoder_outputs[:,:-1,:]
        out_tmp = tf.multiply(output_trim,masks)
        targ_re = tf.multiply(corpt_inputs,masks)
        loss_re = mse_error(out_tmp,targ_re)
        #loss_km
        HTH = tf.matmul(latent_rep,tf.transpose(latent_rep))
        F_copy = tf.get_variable('F', shape=[config.batch_size, config.k_cluster],
                        initializer=tf.orthogonal_initializer(gain=1.0, seed=None, dtype=tf.float32),
                        dtype=tf.float32,
                        trainable=False
                        )

But I can't seem to find anything about loss_km in pypots.crli

**Comment by WenjieDu** [comment]
https://github.com/WenjieDu/PyPOTS/blob/eb02440b491413de2f29351d590bcf8216cc4bd4/pypots/clustering/crli/modules/core.py#L106

**Reply by PBemmm** [reply]
I got it, thank you

**Reply by WenjieDu** [reply]
@PBemmm No problem. You can follow me on GitHub to receive our latest open-source research. We have interesting work about clustering to be published this year ;-)
